### PR TITLE
bazel: Bump version to 0.21.0

### PR DESCRIPTION
**Pull request by rubendibattista**

### CODE REVIEW

**Comment by rubendibattista** [comment]
Ehi @tgamblin, I have no idea. I just copied what what there before to make it build. I don't even know why that is needed, I'm not very fluent in Java (and I'm not a lover :)). 

First thing: why those env vars are needed?

**Comment by rubendibattista** [comment]
I guess it would be cool to upstream the env var handling in order to avoid patching the java files... But I have no idea on where to start.

**Comment by scheibelp** [comment]
> First thing: why those env vars are needed?

The environment variables are how Spack passes information to the compiler wrappers that it inserts into whatever build system is being used by the package (in this case Bazel). For example, some of the environment variables are used to augment a given compiler invocation to add `-I` and `-L` entries to dependencies that were installed by Spack.

**Comment by scheibelp** [comment]
> is there a more standard way to pass environment variables through like this

@tgamblin Bazel is something of a special case. Some users implement patches for "in-flux" files using Spack's `FileFilter`. I'm inclined to say that just updating the patch is conceptually simpler, although if the Spack environment variables change, it means more work updating the patches. For now this seems acceptable.

File: var/spack/repos/builtin/packages/bazel/unix_cc_configure-0.21.0.patch [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+         inc_dirs = result.stderr[index1 + 1:index2].strip()
+ 
+-    return [
++    default_inc_directories = [
+         _prepare_include_path(repository_ctx, _cxx_inc_convert(p))
+         for p in inc_dirs.split("\n")
+     ]
+ 
++    env = repository_ctx.os.environ
++    if "SPACK_DEPENDENCIES" in env:
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/bazel/unix_cc_configure-0.21.0.patch]: @rubendibattista what Spack version are you working with? The latest develop version does not use `SPACK_DEPENDENCIES` so this won't have an effect. There is a `SPACK_INCLUDE_DIRS` variable.

**Pull request closed without merging**